tools/JoinPaths.cmake

```diff
@@ -5,19 +5,21 @@
 # Copyright 2020 Jan Tojnar
 # https://github.com/jtojnar/cmake-snips
 #
-# Modelled after Python’s os.path.join
+# Modelled after Python's os.path.join
 # https://docs.python.org/3.7/library/os.path.html#os.path.join
 # Windows not supported
 function(join_paths joined_path first_path_segment)
-    set(temp_path "${first_path_segment}")
-    foreach(current_segment IN LISTS ARGN)
-        if(NOT ("${current_segment}" STREQUAL ""))
-            if(IS_ABSOLUTE "${current_segment}")
-                set(temp_path "${current_segment}")
-            else()
-                set(temp_path "${temp_path}/${current_segment}")
-            endif()
-        endif()
-    endforeach()
-    set(${joined_path} "${temp_path}" PARENT_SCOPE)
+  set(temp_path "${first_path_segment}")
+  foreach(current_segment IN LISTS ARGN)
```

Comment:
Is there some way to tell pre-commit.ci that this file belongs to another repo and shouldn't be autoformatted as that will result in a confusing file delta between here and the upstream version? :eyes:

Reply:
Yes, you can add an exclude key regex to the pre-commit-config.yaml for the relevant hook.

Reply:
Filename seems to have a dot in front of it...

This file is very big and I have basically zero knowledge of how it works other than "frustrating thing that messes up my commit messages", can I have a small hint at what I'm looking at, here? :)

Reply:
https://github.com/pyodide/pyodide/blob/be1b3c2a6f3ae869300b5630f523dbfb15e3d2e3/.pre-commit-config.yaml#L1

is an example.

Reply:
Also could be worth mentioning this file can be removed once we depend on 3.20+.

Reply:
Ah, thanks. Global exclude sounds helpful.

Reply:
I added comments in CMakeLists.txt, marked with `# TODO: ` for easy searchability.